src/ser.rs

```diff
@@ -11,22 +11,25 @@ use crate::py_type::PyTypeLookup;
 pub struct SerializePyObject<'py> {
     obj: &'py PyAny,
     py: Python<'py>,
+    none_value: &'py str,
     ob_type_lookup: &'py PyTypeLookup,
 }
 
 impl<'py> SerializePyObject<'py> {
-    pub fn new(py: Python<'py>, obj: &'py PyAny) -> Self {
+    pub fn new(py: Python<'py>, obj: &'py PyAny, none_value: &'py str) -> Self {
         Self {
             obj,
             py,
+            none_value,
             ob_type_lookup: PyTypeLookup::cached(py),
         }
     }
 
-    fn with_obj(&self, obj: &'py PyAny) -> Self {
+    fn with_obj(&self, obj: &'py PyAny, none_value: &'py str) -> Self {
         Self {
             obj,
             py: self.py,
+            none_value,
```

Comment:
```suggestion
    fn with_obj(&self, obj: &'py PyAny) -> Self {
        Self {
            obj,
            py: self.py,
            none_value: self.none_value,
```

Then remove the argument in below calls.